pkg/controller/sesv2/emailidentity/setup.go

```diff
@@ -0,0 +1,261 @@
+	configurationSetAttributesInput := &svcsdk.PutEmailIdentityConfigurationSetAttributesInput{
+		ConfigurationSetName: cr.Spec.ForProvider.ConfigurationSetName,
+		EmailIdentity:        cr.Spec.ForProvider.EmailIdentity,
+	}
+	if _, err := e.client.PutEmailIdentityConfigurationSetAttributesWithContext(ctx, configurationSetAttributesInput); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+	}
+
+	// Update MailFromAttributes (PutEmailIdentityMailFromAttributes)
+	// NOTE: Currently MailFromAttributes is not supported in aws-sdk-go v1.44.174 and is declared in Custom-Type
+	if cr.Spec.ForProvider.MailFromAttributes != nil {
+		mailFromAttributesInput := &svcsdk.PutEmailIdentityMailFromAttributesInput{
+			BehaviorOnMxFailure: cr.Spec.ForProvider.MailFromAttributes.BehaviorOnMxFailure,
+			EmailIdentity:       cr.Spec.ForProvider.EmailIdentity,
+			MailFromDomain:      cr.Spec.ForProvider.MailFromAttributes.MailFromDomain,
+		}
+
+		if _, err := e.client.PutEmailIdentityMailFromAttributesWithContext(ctx, mailFromAttributesInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
+		}
+	}
+
+	// Update DKIMSigningAttributes (PutEmailIdentityDkimSigningAttributes)
+	// NOTE: Should be triggered by isUpToDateDkimSigningAttributes(cr, resp)
+	// TODO(kelvinwijaya): This is temporary solution until aws-sdk-go version upgrade
+	dkimSigningAttributeExternal := "EXTERNAL" // Ref: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-authentication-dkim-bring-your-own.html
+	dkimSigningAttributeEasyDKIM := "AWS_SES"  // Ref: https://docs.aws.amazon.com/ses/latest/dg/send-email-authentication-dkim-easy.html
+	if cr.Spec.ForProvider.DkimSigningAttributes != nil && cr.Spec.ForProvider.DkimSigningAttributes.DomainSigningSelector != nil && cr.Spec.ForProvider.DomainSigningPrivateKeySecretRef != nil {
+		// Retrieve from Secret Object if DomainSigningPrivateKeySecretRef is available
+		pk, err := e.getPrivateKeyFromRef(ctx, cr.Spec.ForProvider.DomainSigningPrivateKeySecretRef)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "private key retrival failed")
+		}
+		if len(pk) == 0 {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "private key is not supplied")
+		}
+
+		dkimSigningAttributesInput := &svcsdk.PutEmailIdentityDkimSigningAttributesInput{
+			EmailIdentity: cr.Spec.ForProvider.EmailIdentity,
+			SigningAttributes: &svcsdk.DkimSigningAttributes{
+				DomainSigningPrivateKey: awsclient.String(base64.StdEncoding.EncodeToString([]byte(pk))),
+				DomainSigningSelector:   cr.Spec.ForProvider.DkimSigningAttributes.DomainSigningSelector,
+			},
+			SigningAttributesOrigin: awsclient.String(dkimSigningAttributeExternal),
+		}
+		if _, err := e.client.PutEmailIdentityDkimSigningAttributesWithContext(ctx, dkimSigningAttributesInput); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, "update failed")
```

Comment:
There should be a different error message for each `Wrap` to help identify where the error came from. `"update failed"` is too generic here.

Reply:
resolved